crates/goose/src/agents/truncate.rs

```diff
@@ -343,7 +359,13 @@ impl Agent for TruncateAgent {
 
                                 // Add pre-approved tools
                                 for (request_id, tool_call) in approved_tools {
-                                    let tool_future = Self::create_tool_future(&capabilities, tool_call, request_id.clone());
+                                    let tool_future = Self::create_tool_future(
```

Comment:
the main challenge with this approach is that we don't want token counting to happen in the agent loop cause that is slow. instead its preferred to wait till we get the context length exceeded error and then handle the logic. would it be possible to do this check here: https://github.com/block/goose/blob/36a8be4c395134504a2717c70507c6408c04ab60/crates/goose/src/agents/truncate.rs#L486-L516